### PR TITLE
add init -reconfigure flag

### PR DESCRIPTION
**Pull request by jbardin**
The reconfigure flag will force init to ignore any saved backend state.
This is useful when a user does not want any backend migration to
happen, or if the saved configuration can't be loaded at all for some
reason.

### CODE REVIEW

File: command/init.go [hunk 1]
```diff
@@ -234,9 +239,7 @@ Options:
 
   -no-color            If specified, output won't contain any color.
 
-  -force-copy          Suppress prompts about copying state data. This is
-                       equivalent to providing a "yes" to all confirmation
-                       prompts.
+  -reconfigre          Reconfigure the backend, ignoring any saved configuration.
```
Comment (apparentlymart) [inline command/init.go]: There's a typo in the option name here

File: website/source/docs/commands/init.html.markdown [hunk 2]
```diff
@@ -60,8 +63,7 @@ The command-line flags are all optional. The list of available flags are:
 
 * `-no-color` - If specified, output won't contain any color.
 
-* `-force-copy` -  Suppress prompts about copying state data. This is equivalent
-  to providing a "yes" to all confirmation prompts.
+* `-reconfigre` - Reconfigure the backend, ignoring any saved configuration.
```
Comment (apparentlymart) [inline website/source/docs/commands/init.html.markdown]: Also typo here

Comment (jbardin) [inline website/source/docs/commands/init.html.markdown]: Thanks, I'm an expert at cut&paste errors ;)

**Comment by jbardin** [comment]
It is in essence "reconfiguring" since it currently acts as if the user removed `.terraform/terraform.tfstate` before running init (though it's possible in the future that other info in that "state" file could be used). 

I also can't think of a better name, maybe `-reconfigure-backend`, though that seems a little long?

**Comment by apparentlymart** [comment]
After mulling this some more the only other idea I came up with was the rather-more-literal `-ignore-existing-backend`, which is also quite long. But perhaps this should be long because it's something that should rarely be used?

I don't feel strongly about the name, anyway. If you're happy with `-reconfigure` then feel free to ignore me. :grinning:

**Comment by jbardin** [comment]
Yeah, I didn't want to put "force" or "ignore" in there, because I was sort of leaving this open for adding some sanity checks if they become required, which means we might load the saved config, or add arguments to the `-reconfigure` flag.

**Pull request merged**